### ISSUE TITLE
CI examples and tests are failing with 403: Forbidden

### ISSUE THREAD

**Issue by metral** [pulumi/pulumi-eks#244]
### Problem description

CI examples and tests are failing due to `403: Forbidden` errors on the `testPreviewUpdateAndEdits()`.

https://github.com/pulumi/pulumi/blob/6ea8b3b3dd501460566066bccea8e082421d5031/pkg/testing/integration/program.go#L765-L767

### Errors & Logs

```
[ examples/tests/tag-input-types ] Permalink: https://app.pulumi.com/moolumi/test-tag-input-types/p-it-travis-job-tag-input--375be9bd/previews/ae0ac5a6-a16d-4298-bde3-2b090576d471
[ examples/tests/tag-input-types ] error: failed to complete update: [403] Forbidden
```

@chrsmith's explanation on Slack of 403's from earlier today seems plausible here given the increase in network traffic caused by the doubling of tests due to `RunUpdateTest` being set on all examples and tests:

>[The 403 error is] because the update’s lease token had expired. The CLI needs to request a new one every few minutes. We’ve seen this failing to happen if “there is a lot of network traffic” and the request gets starved or the Goroutine making it gets starved.

Related: https://github.com/pulumi/pulumi-eks/issues/243.

### Reproducing the issue

It seems that enabling the testing framework option [`RunUpdateTest`](https://github.com/pulumi/pulumi/pull/3110) on all examples and tests is the way to repro this, as it doubles the number of tests and network traffic.

**Comment by metral** [comment]
The info detailed in https://github.com/pulumi/pulumi/issues/3182#issuecomment-533706476 confirms our thoughts that the load of doubling the tests is in part related to the instance our CI runs in Travis being limited on resources.

We should consider either more capable Travis instances if applicable, or an alternative to Travis.

**Comment by lukehoban** [comment]
@metral What work is this tracking currently?  Is it a dupe of https://github.com/pulumi/pulumi/issues/3182 or is it something else we need to do?

**Comment by metral** [comment]
https://github.com/pulumi/pulumi/issues/3182 was a spike that stemmed from seeing this issue after the increased load of enabling & running the `RunUpdateTests` on **all** EKS examples && tests.

Per our offline discussion last week, we decided to at least enable the update tests on the core examples to make use of the work that went into implementing `RunUpdateTests`, which we are tracking in PR https://github.com/pulumi/pulumi-eks/pull/260.

This issue only manifests itself when the network load is really heavy, and we have not seen that downstream when testing #260.

**Comment by metral** [comment]
Closed with #260
